frontend/src/components/MonacoEditor/lineDecorations.ts

```diff
@@ -0,0 +1,115 @@
+import * as monaco from "monaco-editor";
+import { ref } from "vue";
+
+const DELIMITER = ";";
+
+type FragmentRange = {
+  startLineNumber: number;
+  startColumn: number;
+  endLineNumber: number;
+  endColumn: number;
+};
+
+/**
+ * Get the ranges of SQL fragments split by the delimiter
+ * @param editor monaco.editor.IStandaloneCodeEditor
+ * @returns fragmentRangeList FragmentRange[]
+ */
+const getSQLFragmentRangeList = (
+  editor: monaco.editor.IStandaloneCodeEditor
+): FragmentRange[] => {
+  const model = editor.getModel() as monaco.editor.ITextModel;
+  const linesContent = model.getLinesContent();
+  const sqlFragmentRangeList: FragmentRange[] = [];
+  let startLineNumber = 0;
+
+  for (let i = 0; i < linesContent.length; i++) {
+    const content = linesContent[i];
+    const range: FragmentRange = new monaco.Range(0, 0, 0, 0);
+
+    const idx = content.indexOf(DELIMITER);
+    // whether the content of line is a empty content
+    const isEmptyContent = monaco.Range.isEmpty(
+      new monaco.Range(i, 0, i, content.length)
+    );
+
+    // first non-empty line
+    if (!isEmptyContent && startLineNumber === 0) {
+      startLineNumber = i + 1;
+    }
+
+    // which contains the delimiter
+    if (idx !== -1) {
+      range.startLineNumber = startLineNumber;
+      // startColumn is always first character of line
+      range.startColumn = 1;
+      // endLineNumber is the line number contains the delimiter
+      range.endLineNumber = i + 1;
+      // keep the last character of the line in the range
+      range.endColumn = idx + 2;
+
+      // jump to the next line of end line
+      startLineNumber = range.endLineNumber + 1;
+
+      sqlFragmentRangeList.push(
+        new monaco.Range(
+          range.startLineNumber,
+          range.startColumn,
+          range.endLineNumber,
+          range.endColumn
+        )
+      );
+    }
+
+    // continue count the empty content
+    if (isEmptyContent && startLineNumber > 0) {
+      startLineNumber += 1;
+    }
+  }
+  return sqlFragmentRangeList;
+};
+
+const lineDecorations = ref<string[]>([]);
```

Comment:
List

Reply:
> List

For custom functions/variables, I agree with that.

Because we use some lib's API, the name of the API is xxxs.

So, I want to keep the style of the API name.

Otherwise, there may well be a day when I can't figure out what API I'm using based on xxxList

<img width="856" alt="CleanShot 2022-03-16 at 13 07 13@2x" src="https://user-images.githubusercontent.com/6118824/158520595-634c681e-2d1a-4630-ab92-ae90be869e17.png">

Reply:
I see.